src/pkg/bb/bbinternal/bb.go

```diff
@@ -471,7 +472,16 @@ func (p *Package) rewriteFile(f *ast.File) bool {
 	hasMain := false
 
 	// Change the package name declaration from main to the command's name.
-	f.Name.Name = p.Name
+	// Remove all non-alphanumeric characters except for underscore and ensure
+	// starting with a letter. There are more valid identifiers though.
+	// The Go spec defines the following grammar:
+	//   identifier = letter { letter | unicode_digit } .
+	// See also https://golang.org/ref/spec#Identifiers
+	reg, err := regexp.Compile("[^a-zA-Z0-9_]+")
```

Comment:
unlikely, but can we get any unicode character here? i'd say... optional

Reply:
Yea, it is not that likely... I left the comment just in case. I think it's fine otherwise to just keep alphanumerics and `_`.